bundles/org.openhab.automation.jsscripting/src/main/java/org/openhab/automation/jsscripting/internal/fs/watch/JSDependencyTracker.java

```diff
@@ -31,32 +32,23 @@
  * Tracks JS module dependencies
  *
  * @author Jonathan Gilbert - Initial contribution
+ * @author Jan N. Klug - Refactored to new WatchService
  */
 @Component(service = JSDependencyTracker.class)
 @NonNullByDefault
 public class JSDependencyTracker extends AbstractScriptDependencyTracker {
 
     private final Logger logger = LoggerFactory.getLogger(JSDependencyTracker.class);
 
-    public static final String LIB_PATH = String.join(File.separator, OpenHAB.getConfigFolder(), "automation", "js",
-            "node_modules");
-
-    public JSDependencyTracker() {
-        super(LIB_PATH);
-    }
+    public static final String LIB_PATH = String.join(File.separator, "automation", "js", "node_modules");
 
     @Activate
-    public void activate() {
-        File directory = new File(LIB_PATH);
-        if (!directory.exists()) {
-            if (!directory.mkdirs()) {
-                logger.warn("Failed to create watched directory: {}", LIB_PATH);
```

Comment:
Does the new watch service in core create the directory if it doesn't exist?
Ignore this question, I just found the code to do this in your core PR in the `WatchServiceImpl`.